Atividades/Em casa/alexandra-ribeiro/clinicaVet.js

```diff
@@ -0,0 +1,204 @@
+class HistoricoMedico {
+    constructor() {
+        this.vacinas = [];
+        this.consultas = [];
+    }
+}
+
+class Animal {
+    constructor(nome, idade, cor) {
+        this.nome = nome;
+        this.idade = idade;
+        this.cor = cor;
+        this.historico = new HistoricoMedico();
+    }
+
+    brincar() { console.log(`${this.nome} está brincando`)}
+
+    consultar() {
+        this.historico.consultas.push(new Date())
+    }
+
+    acariciar() { console.log(`${this.nome} foi acariciado`)}
+
+    alimentar() { console.log(`${this.nome} foi alimentado`)}
+    
+}
+
+class Gato extends Animal {
+    constructor(nome, idade, cor, castrado, externo, social) {
+        super(nome, idade, cor)
+        this.castrado = castrado;
+        this.externo = externo;
+        this.social = social;
+    }
+
+    miar() { console.log("miau") }
+
+    vacinar(vacina) {
+        this.historico.vacinas.push(vacina)
+        this.consultar()
+    }
+
+    consultar() {
+        super.consultar();
+    }
```

Comment:
uma dica é que se você não for sobrescrever um método da classe animal, não precisa chamar ele no gato, pois o gato estende os métodos também, você pode dar por exemplo: nida.consultar() que da certinho.